src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,482 @@
+the [`CholeskySpace`](@ref) as
+````math
+\log_xy = lw\mathrm{T} + wl^{\mathrm{T}},
+````
+where $l$ is the colesky factor of $x$ and $w=\log_lk$ for $k$ the cholesky factor
+of $y$ and the just mentioned logarithmic map is the one on [`CholeskySpace`](@ref).
+"""
+function log(M::MetricManifold{SymmetricPositiveDefinite{N},LogCholeskyMetric},v,x,y) where N
+    l = cholesky(x).L
+    k = cholesky(y).L
+    log!(CholeskySpace{N}(), v, l, k)
+    tangent_cholesky_to_tangent_spd!(l, v)
+    return v
+end
+@doc doc"""
+    representation_size(M)
+
+returns the size of an array representing an element on the
+[`SymmetricPositiveDefinite`](@ref) manifold `M`,
+i.e. $n\times n$, the size of such a symmetric positive definite matrix on
+$\mathcal M = \mathcal P(n)$.
+"""
+representation_size(::SymmetricPositiveDefinite{N}) where N = (N,N)
+
+@doc doc"""
+    vector_transport_to(M,vto,x,v,y,m::AbstractVectorTransportMethod=ParallelTransport())
+
+compute the vector transport on the [`SymmetricPositiveDefinite`](@ref) with its
+default metric, [`LinearAffineMetric`](@ref) and method `m`, which defaults to [`ParallelTransport`](@ref).
+"""
+vector_transport_to!(M::SymmetricPositiveDefinite{N},vto, x, v, y, m::AbstractVectorTransportMethod) where N = vector_transport_to!(MetricManifold(M,LinearAffineMetric()),vto, x, v, y, m)
+
+@doc doc"""
+    vector_transport_to!(M,vto,x,v,y,::ParallelTransport)
+
+compute the parallel transport on the [`SymmetricPositiveDefinite`](@ref) as a
+[`MetricManifold`](@ref) with the [`LinearAffineMetric`](@ref).
+The formula reads
+
+```math
+P_{x\to y}(v) = x^{\frac{1}{2}}
+\operatorname{Exp}\bigl(
+\frac{1}{2}x^{-\frac{1}{2}}\log_x(y)x^{-\frac{1}{2}}
+\bigr)
+x^{-\frac{1}{2}}v x^{-\frac{1}{2}}
+\operatorname{Exp}\bigl(
+\frac{1}{2}x^{-\frac{1}{2}}\log_x(y)x^{-\frac{1}{2}}
+\bigr)
+x^{\frac{1}{2}},
+```
+
+where $\operatorname{Exp}$ denotes the matrix exponential
+and `log` the logarithmic map on [`SymmetricPositiveDefinite`](@ref)
+(again with respect to the metric mentioned).
+"""
+function vector_transport_to!(M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, vto, x, v, y, ::ParallelTransport) where N
+    if distance(M,x,y)<2*eps(eltype(x))
+        copyto!(vto, v)
+        return vto
+    end
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = sqrt.(S)
+    SsqrtInv = Diagonal( 1 ./ Ssqrt )
+    Ssqrt = Diagonal( Ssqrt )
+    xSqrt = Symmetric(U*Ssqrt*transpose(U))
+    xSqrtInv = Symmetric(U*SsqrtInv*transpose(U))
+    tv = Symmetric(xSqrtInv * v * xSqrtInv)
+    ty = Symmetric(xSqrtInv * y * xSqrtInv)
+    e2 = eigen( ty )
+    Se = Diagonal( log.(e2.values) )
+    Ue = e2.vectors
+    ty2 = Symmetric(Ue*Se*transpose(Ue))
+    e3 = eigen( ty2 )
+    Sf = Diagonal( exp.(e3.values) )
+    Uf = e3.vectors
+    xue = xSqrt*Uf*Sf*transpose(Uf)
+    vtp = xue * ( 0.5*(tv + transpose(tv)) ) * transpose(xue) #symmetrize
+    copyto!(vto, vtp)
```

Comment:
Multiply directly into `vto` via
```julia
mul!(vto, xue, Symmetric(tv) * transpose(xue))
```
The second allocates, but probably there's no way around.

Reply:
Hm, interestingly this one line is also the same as above (with `mul!` and `rmul!` there) – exchanging the copy with what you propose breaks the tests – as above I do not know why.

Reply:
Even more interestingly: replacing `vtp = Symmetric(xue*tv*transpose(xue))` _ does_ work, while the `mul!` does not? Maybe I am just getting fully confused.